pkg/action/archive.go

```diff
@@ -257,17 +258,46 @@ func extractArchiveToTempDir(ctx context.Context, path string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to read directory: %w", err)
 	}
-
 	for _, file := range files {
-		extractedFiles[filepath.Base(file.Name())] = false
+		extractedFiles[filepath.Join(tmpDir, file.Name())] = false
 	}
 
 	for file := range extractedFiles {
 		ext := getExt(file)
-		if _, ok := archiveMap[ext]; ok {
-			if err := extractNestedArchive(ctx, tmpDir, file, extractedFiles); err != nil {
-				return "", fmt.Errorf("extract nested archive: %w", err)
+		info, err := os.Stat(file)
+		if err != nil {
+			return "", fmt.Errorf("failed to stat file: %w", err)
+		}
+		switch mode := info.Mode(); {
+		case mode.IsDir():
+			err = filepath.WalkDir(file, func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				rel, err := filepath.Rel(tmpDir, path)
+				if err != nil {
+					return fmt.Errorf("filepath.Rel: %w", err)
+				}
+				if !d.IsDir() {
+					if err := extractNestedArchive(ctx, tmpDir, rel, extractedFiles); err != nil {
+						return fmt.Errorf("extract nested archive: %w", err)
```

Comment:
Perhaps we could add the path here to have more context information.

Reply:
Added in [`0c17487` (#360)](https://github.com/chainguard-dev/bincapz/pull/360/commits/0c1748726cd4820b4023ad04040c5987a776d322)!